pytorch_lightning/accelerators/ddp_accelerator.py

```diff
@@ -133,11 +133,13 @@ def _call_children_scripts(self):
                 del env_copy['PL_GLOBAL_SEED']
 
             # start process
-            # if hydra is available and initialized, make sure to set the cwd correctly
+            # if hydra is available and initialized, make sure to set the original cwd correctly
+            # and pass current cwd for ddp processes (which hydra has overridden)
             cwd: Optional[str] = None
             if HYDRA_AVAILABLE:
                 if HydraConfig.initialized():
                     cwd = get_original_cwd()
+                    command += [f'hydra.run.dir={os.getcwd()}']
```

Comment:
You can use a different .hydra dir for each subjob so they don't overstep one another (or disable it completely.
currently, the subjobs will overwrite the files in the [.hydra](https://hydra.cc/docs/next/tutorials/basic/running_your_app/working_directory/) dir generated by the parent.

Disable:
```
hydra.output_subdir=null
```
Or use your local rank variable to give each subjob it's own .hydra dir:
```
hydra.output_subdir=.hydra_{local_rank}
```

By default, the log filename is derived from the job name. you can override that to something that depends on your local_rank to ensure that each subprocess gets their own job. (I didn't try it here but it should work):
```
hydra.job.name=subjob_{local_rank}
```

Reply:
This is perfect, was my last worry! Will make this change and add a test